flowbased-computation/flowbased-computation-impl/src/main/java/com/farao_community/farao/flowbased_computation/impl/FlowbasedComputationImpl.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.farao_community.farao.flowbased_computation.impl;
+
+import com.farao_community.farao.commons.Unit;
+import com.farao_community.farao.data.crac_api.Cnec;
+import com.farao_community.farao.data.crac_api.Contingency;
+import com.farao_community.farao.data.crac_api.Crac;
+import com.farao_community.farao.data.crac_api.State;
+import com.farao_community.farao.data.flowbased_domain.*;
+import com.farao_community.farao.flowbased_computation.*;
+import com.farao_community.farao.data.glsk.import_.glsk_provider.GlskProvider;
+import com.farao_community.farao.sensitivity_computation.SystematicSensitivityInterface;
+import com.farao_community.farao.sensitivity_computation.SystematicSensitivityResult;
+import com.farao_community.farao.commons.RandomizedString;
+import com.google.auto.service.AutoService;
+import com.powsybl.computation.ComputationManager;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.sensitivity.factors.variables.LinearGlsk;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+/**
+ * Flowbased computation implementation
+ *
+ * @author Sebastien Murgey {@literal <sebastien.murgey at rte-france.com>}
+ */
+@AutoService(FlowbasedComputationProvider.class)
+public class FlowbasedComputationImpl implements FlowbasedComputationProvider {
+    private static final Logger LOGGER = LoggerFactory.getLogger(FlowbasedComputationImpl.class);
+
+    @Override
+    public String getName() {
+        return "SimpleIterativeFlowBased";
+    }
+
+    @Override
+    public String getVersion() {
+        return "1.0.0";
+    }
+
+    @Override
+    public CompletableFuture<FlowbasedComputationResult> run(Network network, Crac crac, GlskProvider glskProvider, ComputationManager computationManager, String workingVariantId, FlowbasedComputationParameters parameters) {
+        Objects.requireNonNull(network);
+        Objects.requireNonNull(crac);
```

Comment:
You should maybe synchronize the crac somewhere, shouldn't you? Otherwise you might get a com.farao_community.farao.data.crac_impl.NotSynchronizedException, if the crac contains relative thresholds.

Reply:
Or when the threshold is in A by the way. One gets "FlowThreshold unit conversion : voltage level must be synchronised with a Network". True story.

Reply:
I don't know if we should do it in the model or in the client call but yeah, it must be somewhere